data-solutions/gcs-to-bq-with-least-privileges/main.tf

```diff
@@ -0,0 +1,245 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+locals {
+  data_eng_users_iam = [
+    for item in var.data_eng_users :
+    "user:${item}"
+  ]
+
+  data_eng_groups_iam = [
+    for item in var.data_eng_groups :
+    "group:${item}"
+  ]
+}
+
+###############################################################################
```

Comment:
I would split each set of resources into a separate file for clarity, e.g. project in `main.tf`, networking resources in `vpc.tf`, etc.